dbt/include/spark/macros/materializations/incremental.sql

```diff
@@ -24,59 +35,79 @@
   {% do return(file_format) %}
 {% endmacro %}
 
+
 {% macro dbt_spark_validate_get_incremental_strategy(file_format) %}
   {#-- Find and validate the incremental strategy #}
-  {%- set strategy = config.get("incremental_strategy", default="insert_overwrite") -%}
+  {%- set strategy = config.get("incremental_strategy", default="append") -%}
 
   {% set invalid_strategy_msg -%}
     Invalid incremental strategy provided: {{ strategy }}
-    Expected one of: 'merge', 'insert_overwrite'
+    Expected one of: 'append', 'merge', 'insert_overwrite'
   {%- endset %}
 
   {% set invalid_merge_msg -%}
     Invalid incremental strategy provided: {{ strategy }}
     You can only choose this strategy when file_format is set to 'delta'
   {%- endset %}
+  
+  {% set invalid_insert_overwrite_delta_msg -%}
+    Invalid incremental strategy provided: {{ strategy }}
+    You cannot use this strategy when file_format is set to 'delta'
+    Use the 'append' or 'merge' strategy instead
+  {%- endset %}
+  
+  {% set invalid_insert_overwrite_endpoint_msg -%}
+    Invalid incremental strategy provided: {{ strategy }}
+    You cannot use this strategy when connecting via endpoint
+    Use the 'append' or 'merge' strategy instead
+  {%- endset %}
 
-  {% if strategy not in ['merge', 'insert_overwrite'] %}
+  {% if strategy not in ['append', 'merge', 'insert_overwrite'] %}
     {% do exceptions.raise_compiler_error(invalid_strategy_msg) %}
   {%-else %}
     {% if strategy == 'merge' and file_format != 'delta' %}
       {% do exceptions.raise_compiler_error(invalid_merge_msg) %}
     {% endif %}
+    {% if strategy == 'insert_overwrite' and file_format == 'delta' %}
+      {% do exceptions.raise_compiler_error(invalid_insert_overwrite_delta_msg) %}
+    {% endif %}
+    {% if strategy == 'insert_overwrite' and target.endpoint %}
+      {% do exceptions.raise_compiler_error(invalid_insert_overwrite_endpoint_msg) %}
+    {% endif %}
   {% endif %}
 
   {% do return(strategy) %}
 {% endmacro %}
 
-{% macro dbt_spark_validate_merge(file_format) %}
-  {% set invalid_file_format_msg -%}
-    You can only choose the 'merge' incremental_strategy when file_format is set to 'delta'
-  {%- endset %}
-
-  {% if file_format != 'delta' %}
-    {% do exceptions.raise_compiler_error(invalid_file_format_msg) %}
-  {% endif %}
-
-{% endmacro %}
-
 
 {% macro spark__get_merge_sql(target, source, unique_key, dest_columns, predicates=none) %}
   {# ignore dest_columns - we will just use `*` #}
+  
+  {% set merge_condition %}
+    {% if unique_key %}
+        on DBT_INTERNAL_SOURCE.{{ unique_key }} = DBT_INTERNAL_DEST.{{ unique_key }}
+    {% else %}
+        on false
```

Comment:
This feels awkward, why not just use an `INSERT INTO` statement?

Reply:
This is in line with default dbt behavior [here](https://github.com/fishtown-analytics/dbt/blob/1508564e10175a7562652f93f08aedb5a2845474/core/dbt/include/global_project/macros/materializations/common/merge.sql#L23-L30).

As I see it, the only reason we implement `spark__get_merge_sql` is to benefit from the improved wildcard `update *`/`insert *`.

I agree it feels a bit silly—this is identical to `append` strategy / `INSERT INTO` statement. The alternative is to keep requiring a `unique_key` with `merge`. In any case, I'd rather err closer to the side of default behavior.